flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarDeserializationSchema.java

```diff
@@ -67,7 +67,13 @@ default void open(InitializationContext context) throws Exception {
      * @param message The message decoded by pulsar.
      * @param out The collector to put the resulting messages.
      */
-    void deserialize(Message<byte[]> message, Collector<T> out) throws Exception;
+    void deserialize(Message<?> message, Collector<T> out) throws Exception;
+
+    /** @return The related Pulsar Schema for this serializer. */
+    default Schema<T> schema() {
```

Comment:
We decided to use try catch to determine if we are using Pulsar Schema , though this is a bit weird as we are using "exception as control flow". Welcome some other options~